src/Tools/SchemaValidator.php

```diff
@@ -276,6 +277,22 @@ public function validateClass(ClassMetadata $class): array
                     }
                 }
             }
+
+            if ($assoc->isIndexed()) {
+                $joinColumns = array_column($targetMetadata->getAssociationMappings(), 'joinColumns');
+                $joinColumns = array_column($joinColumns, 0);
```

Comment:
shouldn't it take *all* join columns into account instead ?

Reply:
I guess it should. I didn't realise/think there could be multiple join columns. Thanks.